<?xml version="1.0"?>
<launch>
  <arg name="amiroID" default="0"/>
  <arg name="meter_per_pixel" value="0.006"/>
  <arg name="pixel_mode" value="False"/>
  <arg name="vectorfield_topic" default="merged_all"/>

  <!-- Odometry to Vector Field -->
  <node name="odom$(arg amiroID)_to_vectorfield_node" pkg="vector_field_navigation" type="odom_to_vectorfield_node" output="screen" cwd="ROS_HOME">
    <param name="meter_per_pixel" value="$(arg meter_per_pixel)"/>
    <param name="amiro_odom_listener_topic" value="/amiro$(arg amiroID)/odom"/>
    <param name="potentialfield_publisher_topic" value="/potentialfield/amiro$(arg amiroID)"/>
    <param name="vectorfield_publisher_topic" value="/vectorfield/amiro$(arg amiroID)"/>
    <param name="heuristic_abs_min" value="0.01"/> <?Undefined An arbitrary value which indecates the start of value decay?>
    <param name="heuristic_factor" value="1.0"/> <?Maximum vector lengh?>
    <param name="minimum_pose_difference_pixel" value="10.0"/> <?Only recalculate the map if the robot has moved that much?>
    <param name="pixel_mode" value="$(arg pixel_mode)"/> <?If this is true pixelmode is activated otherwhise normal mode (metric)?>
    <param name="target_frame" value="vectorfield"/>
  </node>

  <!-- Vector Field to Gridmap -->
  <node name="odom$(arg amiroID)_vectorfield_to_gridmap_node" pkg="vector_field_navigation" type="vectorfield_to_gridmap_node" output="screen" cwd="ROS_HOME">
    <param name="meter_per_pixel" value="$(arg meter_per_pixel)"/>
    <param name="frame_id" value="world"/>
    <param name="vectorfield_listener_topic" value="/vectorfield/amiro$(arg amiroID)"/>
    <param name="gridmap_publisher_topic" value="/vectorfield/amiro$(arg amiroID)/gridmap"/>
    <param name="pixel_mode" value="$(arg pixel_mode)"/> <?If this is true pixelmode is activated otherwhise normal mode (metric)?>
  </node>

  <!-- Vector Field to Kinematic -->
  <node name="amiro$(arg amiroID)_vectorfield_to_kinematic_node" pkg="vector_field_navigation" type="vectorfield_to_kinematic_node" output="screen" cwd="ROS_HOME">
    <param name="meter_per_pixel" value="0.006"/>
    <param name="vectorfield_listener_topic" value="/vectorfield/$(arg vectorfield_topic)"/>
    <param name="amiro_odom_listener_topic" value="/amiro$(arg amiroID)/odom" unless="$(arg pixel_mode)"/>
    <param name="amiro_odom_listener_topic" value="/tracking/pixel/$(arg amiroID)" if="$(arg pixel_mode)" />
    <param name="twist_publisher_topic" value="/amiro$(arg amiroID)/cmd_vel"/>
    <param name="vector_publisher_topic" value="/amiro$(arg amiroID)/vector"/>
    <param name="velocityScale_meterPerSecond" value="0.1"/>
    <param name="angularScale_radPerSecond" value="2.0"/>
    <param name="pixel_mode" value="$(arg pixel_mode)"/>
    <param name="twist_mode" value="True"/>
    <param name="invert_yaw" value="False"/>
    <param name="target_frame" value="vectorfield"/>
  </node>
</launch>
